samples/drivers/spi_flash_at45/src/main.c

```diff
@@ -35,17 +33,19 @@ void main(void)
 	struct flash_pages_info pages_info;
 	size_t page_count, chip_size;
 
-	flash_dev = device_get_binding(FLASH_DEVICE);
+	flash_dev = device_get_binding(CONFIG_SPI_FLASH_AT45_SAMPLE_DEVICE);
 	if (!flash_dev) {
-		printk("Device %s not found!\n", FLASH_DEVICE);
+		printk("Device %s not found!\n",
+		       CONFIG_SPI_FLASH_AT45_SAMPLE_DEVICE);
 		return;
 	}
 
 	page_count = flash_get_page_count(flash_dev);
 	(void)flash_get_page_info_by_idx(flash_dev, 0, &pages_info);
 	chip_size = page_count * pages_info.size;
 	printk("Using %s, chip size: %u bytes (page: %u)\n",
-	       FLASH_DEVICE, chip_size, pages_info.size);
+	       CONFIG_SPI_FLASH_AT45_SAMPLE_DEVICE,
+	       chip_size, pages_info.size);
```

Comment:
Could you also wrap this part in `#if IS_ENABLED(CONFIG_FLASH_PAGE_LAYOUT)`? The `FLASH_PAGE_LAYOUT` option may still happen to be set to `n`.

Reply:
Sure!